parser-typechecker/src/Unison/TermParser.hs

```diff
@@ -225,15 +229,21 @@ parsePattern =
     end <- closeBlock
     pure (Pattern.setLoc inner (ann start <> ann end), vs)
 
+  -- ex: unique type Day = Mon | Tue | ...
+  nullaryCtor = P.try $ do
+    tok <- ctor UnknownAbilityConstructor
+    let (ref, cid) = L.payload tok
+    pure (Pattern.Constructor (ann tok) ref cid [], [])
+
   constructor = do
     tok <- ctor UnknownDataConstructor
     let (ref,cid) = L.payload tok
         f patterns vs =
           let loc = foldl (<>) (ann tok) $ map ann patterns
           in (Pattern.Constructor loc ref cid patterns, vs)
-    unzipPatterns f <$> many patternCandidates
+    unzipPatterns f <$> many leaf
 
-  seqLiteral = Parser.seq f leaf
+  seqLiteral = Parser.seq f root
```

Comment:
There was no reason to restrict the patterns that form the elements of a list pattern to be leaf patterns. Like `[Foo a b]` is a perfectly fine pattern but `Foo a b` is not a `leaf` pattern.